### PR TITLE
[JENKINS-59017] disable ForceSetupWizardFilter if setup complete

### PR DESCRIPTION
**Pull request by BogdanSukonnov**
See [JENKINS-59017](https://issues.jenkins-ci.org/browse/JENKINS-59017).
This change is as minimal as possible. It can't be autotested without massive SetupWizard refactoring. 
I tested this logic by plugin, which adds filter just like FORCE_SETUP_WIZARD_FILTER.
And tested the work of Setup Wizard - OK.

### Proposed changelog entries

* Bug: Sometimes after Setup Wizard completion user saw only blank page

### Submitter checklist

- [X] JIRA issue is well described
- [X] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change).
- [X] Appropriate autotests or explanation to why this change has no tests

### Desired reviewers

@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by BogdanSukonnov** [comment]
@jenkinsci/code-reviewers please review this PR

**Comment by BogdanSukonnov** [comment]
@pengwynn please review this PR

**Comment by oleg-nenashev** [comment]
Will do, sorry for the response delay

**Comment by BogdanSukonnov** [comment]
@oleg-nenashev thanks!

**Comment by Wadeck** [comment]
@BogdanSukonnov Thank you for the bug report and especially the proposed correction. I also encountered the problem multiple times locally but I thought it was my dev setup (stupid hypothesis ^^).

The change seems fine, I am just concerned if it actually cures only the symptoms. The SetupWizard filter should be removed after the end of the setup, by [SetupWizard#tearDownFilter](https://github.com/jenkinsci/jenkins/blob/0e62b68012282648377cfac4fe6d01956d7c1f11/core/src/main/java/jenkins/install/SetupWizard.java#L184). But potentially there is a concurrency issue either [here](https://github.com/jenkinsci/jenkins/blob/0c2d194f0d933fdba8f5a9a1fda8adc8826981c0/core/src/main/java/jenkins/model/Jenkins.java#L1087) or [there](https://github.com/jenkinsci/jenkins/blob/0e62b68012282648377cfac4fe6d01956d7c1f11/core/src/main/java/jenkins/install/SetupWizard.java#L584).

I tried to reproduce the problem voluntarily with `jenkins/jenkins:2.190` (3x with suggested, 3x without any plugin) (released one day before the date you mentionned in the Jira ticket under environment) and with your PR (3x), all attemps were working fine "unfortunately". Even with 2.187 no more success.

**Comment by BogdanSukonnov** [comment]
@Wadeck yep, this is my concern too. And who knows what kind of bugs is related with other undeleted filters? But this problem is too big for me. I'm afraid it needs a major redesign of the core and, possibly, all plugins.
This solution is good enough for now, I think. At list, we can reproduce this problem (add filter in a chain) and see, that this fix works fine.
And if we have the state in application, it's definetely wrong to ignore this state in a filter.

**Comment by BogdanSukonnov** [comment]
@Wadeck 
1 - I can add a debug message to logs, and we can create a new issue about this filter in a chain situation. It will help us to detect this problem, though we don't know how to reproduce the cause.
2 - As far as I understand, main problem with this filter now is that it has not implemented equals. And due to some network or multithread problems the system ends up with more then one ForceSetupWizardFilter. But it deletes only one of them. So maybe we need to also use removeAll  instead of remove first found.  This equals emplementation can be done only with refactoring. I tried to minimize possible mistakes. But I can give it a shot and do the equals implementation. Though it seems pointless unless we can find the way to reproduce main problem.

**Comment by Wadeck** [comment]
@BogdanSukonnov I will try to reproduce with some debugger breakpoint to "force concurrence". Otherwise I am not blocking this PR at all just to be clear, it's good to avoid the issue, I just want to be sure we will also correct the underlying problem, could it be in a separate PR ;)

**Review by Wadeck — APPROVED** [review]
@BogdanSukonnov I can confirm that if you have the blank page problem, it's because you still have (at least) a SetupWizardFilter in place. So yeah your patch will prevent that behavior but for such instances they will keep having that filter completely useless.
IOW PR will solve the problem (hard to prove for sure it's the solution because we could only unprove with a counter-example, but it's very likely as it seems the root cause of the problem is to keep having a filter that will now do nothing).

(FTR the bug only affect the home page, if you try with another page it's working fine)

**Review by oleg-nenashev — APPROVED** [review]
LGTM. Will merge tomorrow if no negative feedback

**Comment by oleg-nenashev** [comment]
Thanks for the patch @BogdanSukonnov !

**Comment by BogdanSukonnov** [comment]
@oleg-nenashev , can you, please set the Issue [https://issues.jenkins-ci.org/browse/JENKINS-59017](url) Resolved?

**Pull request merged**
